deployment/src/lib.rs

```diff
@@ -0,0 +1,208 @@
+extern crate futures;
+extern crate kube;
+extern crate log;
+extern crate thiserror;
+
+use kube::Client;
+use kube::Error as KubeError;
+use serde_yaml::Error as YamlError;
+use serde_json::Error as JsonError;
+use thiserror::Error as ThisError;
+
+use crate::crd::H2OSpec;
+
+pub mod crd;
+pub mod finalizer;
+pub mod ingress;
+pub mod headless_service;
+pub mod statefulset;
+pub mod client;
+
+/// Error during handling Kubernetes cluster-related requests.
+#[derive(ThisError, Debug)]
+pub enum Error {
+    /// Error originating from the Kubernetes API and/or the `kube` crate
+    #[error("Kubernetes reported error: {0}")]
+    KubeError(KubeError),
+    /// Error in user-provided data/configuration
+    #[error("Kubernetes reported error: {0}")]
+    UserError(String),
+    /// Requested operation timed out
+    #[error("Operation timed out. Reason: {0}")]
+    Timeout(String),
+    #[error("Failed to serialize template. Reason: {0}")]
+    TemplateSerializationError(String),
+}
+
+impl Error {
+    /// Constructs a new `deployment::Error::KubeError` with the original error originating from
+    /// the `kube` crate bundled inside.
+    fn from_kube_error(error: KubeError) -> Error {
+        Error::KubeError(error)
+    }
+
+    fn from_serde_yaml_error(error: YamlError) -> Error {
+        Error::TemplateSerializationError(error.to_string())
+    }
+
+    fn from_serde_json_error(error: JsonError) -> Error {
+        Error::TemplateSerializationError(error.to_string())
+    }
+}
```

Comment:
as noted before - individual `From<>` impls might enable be used to enable implicit conversion

Reply:
Implemented `From` traits for each error type.

Reply:
Also deleted the old methods for manual conversion, as those needn't exist at the moment.